### PR TITLE
Add prometheus middleware and server

### PR DESCRIPTION
**Pull request by handou-o**
Added prometheus support based on https://github.com/neo9/n9-node-routing/commit/76b892ecae72121985432adb108e79aa2b978ffa

Needed for Aniel

### CODE REVIEW

File: test/index.ts [hunk 1]
```diff
@@ -111,3 +111,13 @@ test('Fails with PORT not in common range', async (t) => {
 	stdMock.restore()
 	stdMock.flush()
 })
+
+test('Works with prometheus', async (t) => {
```
Comment (BenjD90) [inline test/index.ts]: Could you add a test to check Prometheus metrics page ?

Comment (BenjD90) [reply test/index.ts]: like this one :
https://github.com/neo9/n9-node-routing/commit/76b892ecae72121985432adb108e79aa2b978ffa#diff-eddae6329159783f5cf0a069f102a188R22

File: test/micro-prometheus.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+	t.is(res.body, undefined)
+
+	res = await rp({ uri: 'http://localhost:5000/by-code/code1', resolveWithFullResponse: true, json: true })
+	t.is(res.statusCode, 204)
+	t.is(res.body, undefined)
+
+	// Check /foo route added on foo/foo.init.ts
+	const resProm = await rp({ uri: 'http://localhost:5001/' })
+	t.true(resProm.includes('http_requests_total{method="get",path="/sample-route",status_code="204"} 1'), 'Prom exposition contains call to sample-route')
+	t.true(resProm.includes('http_requests_total{method="get",path="/by-code/code1",status_code="204"} 1'), 'Prom exposition contains call with route pattern')
```
Comment (BenjD90) [inline test/micro-prometheus.ts]: `/by-code/code1` is not the expected metric.
The expected metric is `/by-code/:code`, this is to group similar calls (by default at least).

Comment (handou-o) [reply test/micro-prometheus.ts]: Ok understood as we discussed I added the normalizePath and prepended the test path by /:version(v\\\\d+)?/

**Comment by BenjD90** [comment]
@handou-o could you check the travis check that fail ?

**Pull request merged**